modules/godot_physics_2d/godot_area_2d.cpp

```diff
@@ -283,23 +287,42 @@ void GodotArea2D::call_queries() {
 	}
 }
 
-void GodotArea2D::compute_gravity(const Vector2 &p_position, Vector2 &r_gravity) const {
-	if (is_gravity_point()) {
-		const real_t gr_unit_dist = get_gravity_point_unit_distance();
-		Vector2 v = get_transform().xform(get_gravity_vector()) - p_position;
-		if (gr_unit_dist > 0) {
-			const real_t v_length_sq = v.length_squared();
-			if (v_length_sq > 0) {
-				const real_t gravity_strength = get_gravity() * gr_unit_dist * gr_unit_dist / v_length_sq;
-				r_gravity = v.normalized() * gravity_strength;
+void GodotArea2D::compute_gravity(const Vector2 &p_global_position, Vector2 &r_gravity) const {
+	switch (gravity_type) {
+		case PhysicsServer2D::AREA_GRAVITY_TYPE_DIRECTIONAL: {
+			r_gravity = get_gravity_vector() * get_gravity();
+		} break;
+		case PhysicsServer2D::AREA_GRAVITY_TYPE_POINT:
+		case PhysicsServer2D::AREA_GRAVITY_TYPE_TARGET: {
+			Vector2 target_local_position;
+			if (gravity_type == PhysicsServer2D::AREA_GRAVITY_TYPE_POINT) {
+				target_local_position = get_gravity_vector();
 			} else {
-				r_gravity = Vector2();
+				const Variant local_position_variant = get_inv_transform().xform(p_global_position);
+				const Variant *args[1] = { &local_position_variant };
+				Variant ret;
+				Callable::CallError ce;
+				gravity_target_callback.callp(args, 1, ret, ce);
+				if (ce.error != Callable::CallError::CALL_OK) {
+					ERR_PRINT_ONCE("Error calling event callback method " + Variant::get_callable_error_text(gravity_target_callback, args, 1, ce));
```

Comment:
"event callback" does not point to what could have caused it. You should mention `_calculate_gravity_target()` or something similar that says it happened in Area.